PXC 4132 5.7.41 merge part2

VarunNagaraju opened the pull request.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (fix OL9 RelWithDebInfo build issue)

https://jira.percona.com/browse/PS-8580
***
The RelWithDebInfo build fails on OL9 with [-Werror=stringop-overflow=]
error in field.h::copy_integer.

Backport fix from https://github.com/mysql/mysql-server/commit/9d759f8ff666fc9a625cbcb0bdd657b58c913ca7
to fix the issue.
***
Bug#27962900 RUNTIME ERROR: ADDITION OF UNSIGNED OFFSET TO 0X7F76C0384B89 OVERFLOWED TO 0X7F7

In copy_integer(): don't read more than "from_length" bytes.

GlebShchepa committed:
Bug#29348962: WARNING: POSIX YACC DOES NOT SUPPORT %YACC [-WYACC]

The modern Bison parser generator has started to output warnings
on the usage of some Bison extensions while the %yacc directive
(or the --yacc command line option) is enabled.

The current patch has neutralized that issue by disabling the
warning with --warnings=no-yacc.

Note: --yacc is still enabled, since this is necessary for the
generation of non-conflicting header files. With --yacc,
generated header files contain `#define` directives for each token,
while without --yacc there is the `enum yytokentype` declaration,
and two generated header files can conflict, since both contain
yytokentype.
In modern Bison releases, the name of yytokentype can be altered
with the `%define api.prefix {...}` directive, however releases
prior to 2.6 don't support that feature.

Other changes:

* %yacc directives have beed removed from *.yy files to not
  duplicate the command line option --yacc (there was a Bison
  warning).

* All warnings have been enabled with --warning=all where
  applicable.

Change-Id: Ibee86ae996906dce25f30c755c5f9dfe3ea5bf00

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (backport WITH_PACKAGE_FLAGS to 5.7)

https://jira.percona.com/browse/PS-8580
***
https://github.com/mysql/mysql-server/commit/38474c44e5330c0cbd338ee10613d4d3e703e672#

Bug #32876974 ADD HARDENING FLAGS TO STANDALONE BUILDS ON DEB/RPM PLATFORMS.

RPM and DEB packages typically add CFLAGS and CXXFLAGS to "harden" the
build, e.g. by adding -D_FORTIFY_SOURCE=2.

Add a cmake option WITH_PACKAGE_FLAGS to add the same compiler flags
when doing STANDALONE builds on DEB/RPM platforms.

Set it ON by default for non-DEBUG builds.

Initialize remaining fields in Arg_comparator, to fix:
On Ubuntu 21.04
cmake .. -DWITH_PACKAGE_FLAGS=1 -DWITH_DEBUG=1
sql/item_cmpfunc.h:132:7:
error:  may be used uninitialized in this function [-Werror=maybe-uninitialized]
  132 | class Arg_comparator

Downgrade Werror=format-overflow to warning for some NDBcluster code,
to avoid broken build for:
  -DWITH_NDBCLUSTER=1 -DWITH_PACKAGE_FLAGS=1 -DWITH_DEBUG=1

gkodinov committed:
Implemented PS-8580 (Merge MySQL 5.7.41) (backport Bug#33967647 fix to 5.7)

https://jira.percona.com/browse/PS-8580
***
https://github.com/mysql/mysql-server/commit/0dbc909effc2b85af2194ab4ddf7d53d027b6d02

Bug #33967647: remove ERR_clear_error() from vio_ssl_write and ssl_handshake_loop for openssl3

Ensured openssl ERR queue is cleaned up properly.
Removed the forced clean up pre-check.

bkandasa committed:
Raise version number after cloning 5.6.58

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

bkandasa committed:
Raise version number after cloning 5.7.40

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

Commit:
BUG#33646899: MTS + STOP SLAVE SQL_THREAD + group of rep events stops
 before 60 sec timeout

 Description:
 ------------
 Issuing STOP SLAVE SQL_THREAD command while sql thread is handling
 a transaction that interacts with non-transaction table caused
 below error:

 2022-08-03T18:01:11.995956Z 122 [Warning] Slave SQL for channel '':
 Request to stop slave SQL Thread received while applying a group that
 has non-transactional changes; waiting for completion of the group
 ... , Error_code: 0
 2022-08-03T18:01:13.015261Z 124 [ERROR] Slave SQL for channel '':
 Fatal error: ... Slave SQL Thread stopped with incomplete event
 group having non-transactional changes. If the group consists
 solely of row-based events, you can try to restart the slave with
 --slave-exec-mode=IDEMPOTENT, which ignores duplicate key, key not
 found, and similar errors (see documentation for details).
 Error_code: 1593

 The replica was supposed to wait 60 seconds(SLAVE_WAIT_GROUP_DONE
 timeout) to wait for the group of events to complete and then force
 the stop of the SQL thread. However the timestamps indicate it
 did not wait a second before it threw the error and stopped
 replication.

 Analysis:
 ------------
 The issue is due to the variable(last_event_start_time) which keeps
 last event start time(last_event_start_time) is not getting resetted
 after sql thread restart. When the variable is not reset, and the
 thread is started again, and then stopped a second time, the second
 'stop' will not set a new value from the variable. So it will use
 the same timeout as the preceding 'stop slave', i.e., force-stop
 workers when 60 seconds have elapsed after the *first* stop slave
 statement.The Issue occurs only when running replication in single
 thread mode.

 Fix:
 ----
 Reset last_event_start_time at the start of sql thread.

Change-Id: I3614e89a2d121696970e19f138088567ed068091

Commit:
Bug#34400586 Backport MySQL contribution id: 31497 to 5.7

PROBLEM
-------

1. The page cleaner thread flushes the dirty page in buffer pool
   to the disk.
2. During the shutdown phase the the page cleaner tries to flush
   the dirty pages in a loop, until the number of pages flushed
   is zero. (n_flushed)
3. It is possible that pending reads can cause a ibuf merge which
   causes a dirty page which is not detected until the end of work loop.
4. If this is not detected it can cause further asserts in sanity check
   of the buffer pool

FIX
---
1. Check the pending reads in the buffer pool before exiting the work
   loop.

Change-Id: If313b69af31217e30e9742aa2131af6625d35b39

TharunSai committed:
Bug #34207811 : Audit plugin does not log multiqueries correctly.

Description :
-------------
The Audit plugin does not log multiqueries correctly. It will always log whole
query instead of the actual query that is being executed. This is because the
query length in the THD is set after the general logging is done.

Fix:
----
Setting the query length of the THD before general logging will ensure that only
one query that is being executed is logged.

Note:
-----
* This patch backports the fix for "Bug #31260698 : PERFORMANCE REGRESSION ON
  WINDOWS INTRODUCED BY FIX FOR BUG#30654405".
* This patch also fixes the "Bug #34569180 : P_S.events_statements_history
  tables store wrong data when executing multiqueries".

Change-Id: I1ec7641599deff284ca06cc0357a61d23d33d90d

Commit:
Bug#33799840 - relay_log_space_limit VALID_RANGE improperly defined

Problem
-------
The docs say the limit is 4GB:
https://dev.mysql.com/doc/refman/8.0/en/replication-options-replica.html#option_mysqld_relay-log-space-limit
But it is possible to set much bigger than that:

mysql > select
@relay_log_space_limit/1024/1024/1024,@@version,@@version_comment;
+----------------------------------------+------------+-----------------------
-------+
| @@relay_log_space_limit/1024/1024/1024 | @@version  | @@version_comment
       |
+----------------------------------------+------------+-----------------------
-------+
|                        50.000000000000 | 5.7.35-log | MySQL Community
Server (GPL) |
+----------------------------------------+------------+-----------------------
-------+
1 row in set (0.00 sec)

Analysis / Root-cause analysis
------------------------------
Documentation is written based on MySQL configuration variables declaration.
static Sys_var_ulonglong Sys_relay_log_space_limit(
    "relay_log_space_limit", "Maximum space to use for all relay logs",
    READ_ONLY GLOBAL_VAR(relay_log_space_limit), CMD_LINE(REQUIRED_ARG),
    VALID_RANGE(0, ULONG_MAX), DEFAULT(0), BLOCK_SIZE(1));

Solution
--------
Change Sys_relay_log_space_limit VALID_RANGE to (0, ULLONG_MAX). Also the change modifies valid range to 2^64 on all systems. (ULONG_MAX is 2^32 on 32-bit systems and 2^64 on 64-bit systems).

Change-Id: Idfa55e9cdffb8bb414e45a62482623360bc0b701

TharunSai committed:
Bug #34207811 : Audit plugin does not log multiqueries correctly.

[post-push-fix] Fixing the ASAN failure seen on mysql-5.7

Approved by Tatiana Nuernberg <tatjana.nuernberg@oracle.com> over Slack.

Change-Id: Iaaea6a426011b09d055e7c05a0f26721d0995030

Commit:
Bug #33822493 : pb2 weekly-5.7 test failures due to InnoDB: Assertion
failure

PROBLEM :
- recv_writer_thread is responsible to set variable
  recv_writer_thread_active as True at the start and False at the end.
- Inside recv_recovery_from_checkpoint_finish() server waits for
  recv_writer_thread to finish, by checking the value of
  recv_writer_thread_active.
- Issue arises when scheduling of recv_writer_thread is delayed to
  such an extent where recv_recovery_from_checkpoint_finish()
  considers recv_writer_thread as finished, because
  recv_writer_thread_active is false as a default.
- We hit assertion when recv_writer_thread is scheduled and sets
  recv_writer_thread_active as True and we assert it to be False
  considering it has finished execution.

FIX :
- Setting recv_writer_thread_active as True is done in main thread
  just before the spawn of recv_writer_thread.

Change-Id: I3403dce4fd3071e9adcc8e39ad44bec834769a16

shussinh committed:
Bug# 33630199 - mysqldump make a non-consistent backup with --single-transaction option

Problem
-------
This problem arise when we have a server which has GTIDs enabled, and
there exists few on going transactions in the background or in other sessions.
Basically those kind of transactions which will cause GTIDs to increase.

When executing mysqldump to dump data of this server, with options
--single-transaction and --set-gtid-purged=ON, the dump file generated
has inconsistent data. When examining the dump file, we'll notice that
the dump file has the GTIDs executed for those transactions which were
running in the background, but it does not have the corresponding data
for those GTIDs.

Analysis
--------
When --single-transaction option is specified, a snapshot of the
database at that instant is taken. But the transactions which were
happening in the background are still causing GTIDs to increase.

After a transaction over a database snapshot is started, mysqldump
disables the binlog, unlocks the table, proceeds with the dump, once
dump is over, it fetches the GTIDs. Since GTIDs are being fetched
after the tables were unlocked, GTIDs would have been increased
during the period when dump was running, since many transactions
which were running in the background were completed, resulting in
increment of GTIDs.

Solution
--------
Current flow for mysqldump is as follows:
	1. start transaction
	2. disable binlog
	3. unlock tables
	4. start the dump
	5. dump finish
	6. fetch GTID_EXECUTED, and print it

Here, we see that when the GTIDs are being fetched, tables were
unlocked already, and dump was over too.

Fix does the following:

  1. mysqldump will check if --single-transaction is enabled, if it
  is enabled, mysqldump will perform FTWRL by calling the function
  do_flush_tables_read_lock(mysql);
  This will perform a read lock on the tables.

  2. "my_bool single_trans_with_gtid_enabled" will check if the
  server has GTIDs enabled, and mysqldump was started with the
  option --single-transaction, and --set-gtid-purged is NOT off.

  3. Locks acquired, start the transaction, get the snapshot of the
  database. Now it can fetch the value for GTID_EXECUTED, and store
  it in the variable "MYSQL_RES *gtid_executed". This variable is
  initialized to NULL in the beginning of main().

  ** MYSQL_RES doesn't have a life of its own, it's only valid as
     long as the MYSQL that produced it is in a matching state.
     Hence, it is not preferred to pass the same variable to other
     functions.

     New struct MY_GTID_ARRAY has been introduced, which can hold
     the result of `SELECT @@GLOBAL.GTID_EXECUTED`. It has 2 members,
     one will tell the number of rows, and other one is just an array
     of strings, to store the GTIDs.

     The `if(single_trans_with_gtid_enabled)` block will basically
     perform a deep copy of gtid_executed into the MY_GTID_ARRAY,
     and store result of `SELECT @@GLOBAL.GTID_EXECUTED` into
     gtid_executed.

Thus, this MY_GTID_ARRAY will be passed over to other functions.

Till now, we've both GTIDs executed, and the corresponding data in
the snapshot.

  4. The function responsible for fetching the values of GTIDs
  is

    > add_set_gtid_purged(MYSQL *mysql_con)

  which is being called by the function

    > process_set_gtid_purged(MYSQL* mysql_con, my_bool flag)

  process_set_gtid_purged now takes two more parameter, viz.
  MY_GTID_ARRAY **gtid_array, my_bool single_trans_with_gtid_enabled
  ,and so does the add_set_gtid_purged.

    > process_set_gtid_purged(MYSQL* mysql_con, my_bool flag,
			     MY_GTID_ARRAY **gtid_array,
                             my_bool single_trans_with_gtid_enabled)

    > add_set_gtid_purged(MYSQL *mysql_con,
                          MY_GTID_ARRAY **gtid_array,
                          my_bool single_trans_with_gtid_enabled)

So, basically, the GTIDs which we stored initially after the
FTWRL, are now being passed to this function. And inside of
the process_set_gtid_purged, the same set of GTIDs is being
passed to add_set_gtid_purged.

    - If mysqldump was NOT started with --single-transaction,
    then we did not store GTIDs in the beginning of the dump.

The only condition for this bug is to have a server with gtids
enabled, and a dump started with --single-transaction.
`single_trans_with_gtid_enabled` is keeping track of this condition.

If `single_trans_with_gtid_enabled` is false, then

    add_set_gtid_purged checks if it's false, if it is, it means that
    GTID_EXECUTED wasn't fetched earlier, in that case, run
    the query again to grab GTIDs, then proceed normally. This is
    the normal behaviour, that is, getting the GTIDs after dump.

    Depending on whether `single_trans_with_gtid_enabled` is true
    or false, add_set_gtid_purged will take care of the ways to
    fetch gtids accordingly.

  5. We've now both, snapshot of the data, along with their
  corresponding GTID set.

Earlier versions used to fetch GTIDs at the beginning of the dump,
but the sequence in which GTIDs are fetched has changed because of
the issue caused by Bug# 32843447. Changing the order permanently
in which GTIDs are fetched might trigger that bug again. This is
the reason that only for this particular case the order in which
GTIDs are fetched has changed.

With these changes, the original functionality remains untouched.

Added few more functions

  > static int get_gtid_mode
     This function checks if GTIDs are enabled on the server. And
     return integer result.

  > static my_bool get_gtids_executed
     This function will store the table GTID_EXECUTED in the
     parameter gtid_executed.

  > struct st_my_gtid_array (typedef to MY_GTID_ARRAY)
     A structure to store results for GTID_EXECUTED.

Added new warning in the function : process_set_gtid_purged

  >  Running mysqldump on a GTID enabled server, without
     --single-transaction or --master-data or --lock-all-tables
     will throw a warning about data dump might being inconsistent.

Change-Id: I90347497275fded6a1896b76c6f7a1b04b28fc2e

shussinh committed:
Bug# 33630199 - mysqldump make a non-consistent backup with --single-transaction option

[post-push fix]
Fixing the ASAN failures seen on mysql-5.7 PB2 branch

Approved by: Georgi Kodinov <georgi.kodinov@oracle.com>

Commit:
Revert "Bug#33799840 - relay_log_space_limit VALID_RANGE improperly defined"
This reverts commit 3fa3d4be337b948980f40bda14933f3e54683b8f.

Reviewed-by: Pedro Gomes <pedro.gomes@oracle.com>
Change-Id: I2bfe05ea1a52fb480bc99d5c6e92ec42ef3ce18d

Commit:
Bug#33100586 CRASH IN ITEM_SINGLEROW_SUBSELECT::STORE

Issue:
If the subquery of an order by clause has a view reference, it is
not cleaned up. This is because the subquery might be referred to
again during the execution.

In the case of execution of such a prepared statement, the server
exits as the query expression from the order by clause is present
but not marked optimized.

Fix:
Cleanup query expression of subquery if it is marked as not used
by the resolver.

Change-Id: I7ce2c6243e132339ad69c2c353f64c8d1c0485a2

Commit:
Bug #34809802 Upgrade the bundled lz4 library to version 1.9.4

Upgrade to lz4-1.9.4  This patch is for 5.7

- unpack tarball
- git add LICENCE lib/*.h lib/*.c
- change LZ4_VERSION in cmake/lz4.cmake
- remove old version

Change-Id: Id60237b028f31d15f360b638b34aaee15767de1d

inikep committed:
PS-8502: Building issue with ccache on Azure

1. Remove `##vso[task.prependpath]`.
2. Use `-DCMAKE_C_COMPILER_LAUNCHER=ccache` and `-DCMAKE_CXX_COMPILER_LAUNCHER=ccache`.
3. Use `-DCMAKE_C_COMPILER=$SELECTED_CC` and `-DCMAKE_CXX_COMPILER=$SELECTED_CXX`.
4. Introduce `$PARENT_BRANCH` for the cache string will depend on 5.7 and 8.0 branch.
5. Switch from `ubuntu-18.04` to `ubuntu-20.04` VM images.

bkandasa committed:
Updated copyright year in user visible text

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

bkandasa committed:
Updated copyright year in user visible text

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

inikep committed:
PS-8301: Disable Azure pipelines testing for certain branches

Disable Azure pipelines for branches that don't change source code and don't influece compilation.

bkandasa committed:
Update License Book

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

adivinho committed:
Merge pull request #4939 from percona/release-5.7.40-43

PS-8499 Release 5.7.40-43

inikep committed:
Merge pull request #4940 from inikep/PS-5.7-azure-trigger

PS-8301 [5.7]: Disable Azure pipelines testing for certain branches

inikep committed:
Merge pull request #4938 from inikep/PS-8502-GCA-5.7

PS-8502 [5.7]: Building issue with ccache on Azure

inikep committed:
PS-8502: Separate ccache keys for INVERTED=ON and INVERTED=OFF in Azure Pipelines

inikep committed:
Merge pull request #4943 from inikep/PS-8502-5.7

PS-8502: Separate ccache keys for INVERTED=ON and INVERTED=OFF

inikep committed:
PS-8524: Add gcc-12 compiler to Azure Pipelines

inikep committed:
PS-8524: Fix gcc-12 compilation issues

1. Fix gcc-12 compilation issues.
2. Advance `extra/coredumper`, `plugin/tokudb-backup-plugin/Percona-TokuBackup`, and `storage/tokudb/PerconaFT` submodule pointers.

inikep committed:
Merge pull request #4950 from inikep/PS-8524-5.7

PS-8524 [5.7]: Add gcc-12 to Azure Pipelines and fix gcc-12 compilation issues

TharunSai committed:
Bug #34857411 : regression - slow connections/telnet block many other statements
                for connect_timeout seconds, causing pileups

Description:
------------
This is a regression caused due to the fix made for the Bug 34094706. When a
connection somehow stalls/blocks during the authentication phase, where a mutex
is held, the other connections that are executing queries on I_S and P_S are
blocked until the first connection release the mutex.

Fix:
----
Instead of using the mutex and checking the thd->active_vio, we now check the
value of net.vio type in the is_secure_transport() check.

Change-Id: Ic301a3a69b2bb828f2323441bdd0232e5a0ecef4
(cherry picked from commit c9c360d4e726e79c140855bae7c6f5e7c0c156fb)

inikep committed:
PS-8524: Test all compilers if branch name contains "fullci" with Azure CI

inikep committed:
Merge pull request #4951 from inikep/PS-8524-5.7-fullci

PS-8524 [5.7]: Test all compilers with Azure CI if branch name contains "fullci"

inikep committed:
PS-8546: Add clang-15 to Azure Pipelines

inikep committed:
PS-8546: Fix clang-15 compilation issues

inikep committed:
PS-8546: Fix issues when creating libperconaserverclient.a with /usr/bin/llvm-ar

The issue
```
/usr/bin/ld: ../archive_output_directory/libperconaserverclient.a: error adding symbols: archive has no index; run ranlib to add one
```
was caused by wrong `mri-script` file provided to `$CMAKE_AR`:
```
create /data/mysql-server/percona-5.7-deb-clang15-rocks-toku/archive_output_directory/libperconaserverclient.a
addlib /data/mysql-server/percona-5.7-deb-clang15-rocks-toku/archive_output_directory/libperconaserverclient.a
save
end
```

inikep committed:
Merge pull request #4957 from inikep/PS-8546-5.7-fullci

PS-8546 [5.7]: Add clang-15 to Azure Pipelines and fix clang-15 compilation issues

inikep committed:
PS-8560: Allow to build RocksDB for selected CPU architecture

1. Introduce `ROCKSDB_BUILD_ARCH` parameter to select MyRocks build architecture with `-march=${ROCKSDB_BUILD_ARCH}`.
2. Remove adding instruction set flags e.g. `-msse4.2` to individual files.

inikep committed:
Merge branch '5.7' into PS-8560-5.7

inikep committed:
Merge pull request #4964 from inikep/PS-8560-5.7

PS-8560: Allow to build RocksDB for selected CPU architecture

inikep committed:
PS-8560: Backport improved detection of CPU instruction sets from 8.0

1. Introduce CHECK_VARIABLE_DEFINED cmake function
2. Remove checks for `bmi` and `lzcnt` as they are not used by RocksDB anymore

inikep committed:
Merge pull request #4968 from inikep/PS-8560-5.7-backport

PS-8560 [5.7]: Backport improved detection of CPU instruction sets from 8.0

inikep committed:
PS-8572: Add a cron job to Azure Pipelines

1. Add a daily midnight cron job for 5.7 branch.
2. Do not test git pushes (also merges) to 5.7 branch.
3. Reduce a number of tests for pull requests.

inikep committed:
Merge pull request #4970 from inikep/PS-5.7-cron

PS-8572 [5.7]: Add a cron job to Azure Pipelines

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (docs)

https://jira.percona.com/browse/PS-8580
***
Updated man pages from MySQL Server 5.7.41 source tarball.
***
Updated 'scripts/fill_help_tables.sql' from MySQL Server 5.7.41 source tarball.

inikep committed:
Implemented PS-8580 (Merge MySQL 5.7.41) (extend backup_locks_mysqldump test)

https://jira.percona.com/browse/PS-8580

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41 up to 6e4e344) (merge with conflicts)

https://jira.percona.com/browse/PS-8580

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41 up to 6e4e344) (conflicts resolved)

https://jira.percona.com/browse/PS-8580
***
VERSION raised to "5.7.41-44.0".
PERCONA_INNODB_VERSION in univ.i raised to "44.0".
***
Reverted upstream fix https://github.com/mysql/mysql-server/commit/d2d6b2dd00f709bc528386009150d4bc726e25a0
in favor of PS specific fix https://github.com/percona/percona-server/commit/9bf70e4ce60a847ea28201f80f57e66fddd15ebd.
***
Reverted upstream change https://github.com/mysql/mysql-server/commit/72f28e5ba7838beee4e5eabcf1118024b7b38b2a.
Same issue fixed in the following PS specific changes:
-https://github.com/percona/percona-server/commit/23e08a05f84e8764b15fa64401d56c99720f061f
-https://github.com/percona/percona-server/commit/9e7a981e5109625e15f462edbc3ccb757da91953.
***
Re-record mysql_config_editor test result to reflect actual copyright.

dutow committed:
PS-7538: Crash with innodb_optimize_fulltext_only

Issue: innodb_optimize_fulltext_only=ON allows users to use OPTIMIZE
TABLE to rebuild fulltext indices.

Internally the fulltext rebuild logic reads innodb_ft_num_word_optimize
keys from the index, uses zlib to compress them, and then starts
decompressing/iterating them.

There's an error in this decompressing logic: if zlib reports end of
stream, the reader method assumes that we correctly read the last entry
and returns it.

But it's also possible that we were already at the end of the stream,
and we didn't read anything. The variable used for returning the next word
is reused from the previous iteration in the optimize loop. If it
contains a record that exists in the index, the logic ends up "optimizing"
it again. If it contains a value that's not present in the index, it
crashes.

Fix: only return success if there was actually something in the zip
data.

dutow committed:
PS-7538: Crash with innodb_optimize_fulltext_only

Issue: innodb_optimize_fulltext_only=ON allows users to use OPTIMIZE
TABLE to rebuild fulltext indices.

Internally the fulltext rebuild logic reads innodb_ft_num_word_optimize
keys from the index, uses zlib to compress them, and then starts
decompressing/iterating them.

There's an error in this decompressing logic: if zlib reports end of
stream, the reader method assumes that we correctly read the last entry
and returns it.

But it's also possible that we were already at the end of the stream,
and we didn't read anything. The variable used for returning the next word
is reused from the previous iteration in the optimize loop. If it
contains a record that exists in the index, the logic ends up "optimizing"
it again. If it contains a value that's not present in the index, it
crashes.

Fix: only return success if there was actually something in the zip
data.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (add ASAN suppressions)

https://jira.percona.com/browse/PS-8580

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (fix my_aes_crypt() with OpenSSL 3.0.x)

https://jira.percona.com/browse/PS-8580
***
This is an adapted to PS 5.7 version of the fix from
https://github.com/percona/percona-server/commit/d51913b523a1fda5d35490a701d9c64f86b505ab.

Fixed problem with 'ALTERR TABLE ... ADD FULLTEXT ...' statement
crashing the server when 'innodb_encrypt_online_alter_logs' is
set to 'ON'.

It turned out that the root cause of this problem is Percona-specific
'my_aes_crypt()' function not working as expected with OpenSSL 3.0.x.
Basically, this function cannot properly decrypt data encrypted
previously with the same key/iv. Internally it relies on the value in
the Cipher Context internal buffer ('EVP_CIPHER_CTX_buf_noconst()')
after partial AES update. OpenSSL never claimed that after partial
AES update this value will point to the remainder of the original
message and this behaviour changed in OpenSSL 3.0.x.

Added 'innodb.percona_bug_ps8428' MTR test case which checks
'ALTER TABLE ... ADD FULLTEXT ...' under
'--innodb_encrypt_online_alter_logs=ON'.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (fix ssl_bug75311 test)

https://jira.percona.com/browse/PS-8580
***
Extended replace_result pattern to match new expected error code.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (fix some undefined behaviour errors in item_func)

https://jira.percona.com/browse/PS-8580
***
The main.func_math test fails in Release build with gcc-12. The issue is caused
by undefined behaviour when doing arithmetic with signed operands in item_func.cc.

Partialy backported fixes from https://github.com/mysql/mysql-server/commit/1b4285f9c8dec1fb8968eefc2a22f75f57619600
and https://github.com/mysql/mysql-server/commit/8fcd8fa27970240166b5928e83000e600774d50d.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (remove deprecated Ubuntu-18.04 from Azure builds)

https://jira.percona.com/browse/PS-8580

dutow committed:
Merge pull request #4990 from dutow/ps7538-57

PS-7538: Crash with innodb_optimize_fulltext_only (5.7)

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (fix tlsv1 deprecation tests)

https://jira.percona.com/browse/PS-8580
***
Test framework was not able to shutdown server properly after it is
started without tlsv1.2 enabled.
To fix the issue run include/not_min_protocol_tlsv12.inc before actually
restarting server without tlsv1.2 enabled.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (fix keyring_vault.keyring_udf test)

https://jira.percona.com/browse/PS-8580
***
This test requires both keyring_vault and keyring_udf plugins to be
available at the same time but as they are located in different
directories in source tree we'd have to pass differen locations to
--plugin-dir server option. This is not possible unfortunattely.

To fix the issue both plugins where copied to MYSQLTEST_VARDIR and this
location used as --plugin-dir value.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (fix keyring_vault.keyring_vault_config_qa test)

https://jira.percona.com/browse/PS-8580
***
The '/' path may be opened as a file and on some OSes behaviour may
differ when trying to get this file size.
Add 'Could not read credentials file content' error suppression to fix
test on Fedora Linux.

Commit:
Implemented PS-8580 (Merge MySQL 5.7.41) (no conflicts)

https://jira.percona.com/browse/PS-8580

oleksandr-kachan committed:
Merge pull request #4985 from oleksandr-kachan/merge-mysql-5.7.41-part1

Merge MySQL 5.7.41 (part1)

oleksandr-kachan committed:
Merge pull request #4993 from oleksandr-kachan/merge-mysql-5.7.41-part2

Merge MySQL 5.7.41 (part2)

percona-ysorokin committed:
PS-8580 postfix: Merge MySQL 5.7.41 (keyring_vault_timeout)

https://jira.percona.com/browse/PS-8580

Fixed stability issues with 'keyring_vault.keyring_vault_timeout'
MTR test case caused by sporadic 'no route to host' instead of
'couldn't connect to host' errors reported from CURL.

percona-ysorokin committed:
Merge pull request #4998 from percona-ysorokin/merge-5.7.41-keyring_vault_timeout_fix

PS-8580 postfix: Merge MySQL 5.7.41 (keyring_vault_timeout)

adivinho committed:
PS-8641 Release PS 5.7.41-44

EvgeniyPatlan committed:
Merge pull request #5007 from adivinho/release-5.7.41-44

PS-8641 Release PS 5.7.41-44

adivinho committed:
PS-8641 Release PS 5.7.41-44

adivinho committed:
PS-8641 Release PS 5.7.41-44

adivinho committed:
PS-8641 Release PS 5.7.41-44

adivinho committed:
Merge pull request #5009 from adivinho/release-5.7.41-44

Release 5.7.41 44

adivinho committed:
PS-8655 Dependency issues installing PS on Ubuntu Jammy (#5010)

* PS-8641 fix dependency for jammy

adivinho committed:
Release 5.7.41 44 (#5011)

* PS-8656 Dependency issue in installing PS on oracle linux 9

adivinho committed:
PS-8656 Dependency issue in installing PS on oracle linux 9 (#5012)

* PS-8656 Dependency issue in installing PS on oracle linux 9

adivinho committed:
PS-8656 Dependency issue in installing PS on oracle linux 9

adivinho committed:
Merge pull request #5013 from adivinho/release-5.7.41-44

PS-8656 Dependency issue in installing PS on oracle linux 9

adivinho committed:
PS-8656 Dependency issue in installing PS on oracle linux 9

adivinho committed:
Merge pull request #5015 from adivinho/release-5.7.41-44

PS-8656 Dependency issue in installing PS on oracle linux 9

percona-ysorokin committed:
PS-8580 postfix: Merge MySQL 5.7.41 (native partitioning inline fix)
https://jira.percona.com/browse/PS-8580

Fixed problem with invalid usage of the 'inline' keyword in the
Native Partitioning code causing strange linker errors on Oracle
Linux 9.

percona-ysorokin committed:
Merge pull request #5016 from percona-ysorokin/merge-5.7.41-native_partitioning_inline_fix

PS-8580 postfix: Merge MySQL 5.7.41 (native partitioning inline fix)

VarunNagaraju committed:
Implemented PXC-4132 (Merge PS-5.7.41, Part 1 with conflicts)

https://jira.percona.com/browse/PXC-4132

Merge commit 'b9848e44dcea1a18190cba1d2e5eda84dda29753' into PXC-4132-5.7.41-merge-part1

Conflicts:
        both modified:   MYSQL_VERSION
        both modified:   README
        both modified:   azure-pipelines.yml
        both modified:   storage/tokudb/CMakeLists.txt
        modified:   extra/coredumper (new commits)

Percona-TokuBackup and PerconaFT remains deleted.

VarunNagaraju committed:
Implemented PXC-4132 (Merge PS-5.7.41, Part 1 conflicts resolved)

https://jira.percona.com/browse/PXC-4132

MYSQL_VERSION:
* Bumped PXC_VERSION_EXTRA to 65.
CMakeLists.txt:
* Tokudb remains disabled.
azure-pipelines.yml:
* MacOS platfroms stay deleted from azure pipelines since
  PXC isn't supported yet on MacOS.
* Adds SSL and SSL cmake options.

VarunNagaraju committed:
Implemented PXC-4132 (Merge PS-5.7.41, Part 1 Fixes minor issues)

https://jira.percona.com/browse/PXC-4132

Fixes a compilation failure by creating wsrep_version.cmake to
set WSREP_VERSION which is required by mysql_version.cmake.
Also, bumps the wsrep version.

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1761) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1761) before we can accept your contribution.<br/>**5** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: adivinho<br/>:white_check_mark: inikep<br/>:white_check_mark: VarunNagaraju<br/>:white_check_mark: dutow<br/>:white_check_mark: EvgeniyPatlan<br/>:x: mysql-builder@oracle.com<br/>:x: gipulla<br/>:x: shussinh<br/>:x: percona-ysorokin<br/>:x: bkandasa<br/>:x: oleksandr-kachan<br/>:x: gkodinov<br/>:x: GlebShchepa<br/><hr/>**mysql-builder@oracle.com** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster?pullRequest=1761) it.</sub>

VarunNagaraju commented:
Jenkins results:
https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/381/#showFailuresLink : No new failures.
(https://github.com/Percona-Lab/jenkins-pipelines/pull/2013 fixes the OEL9 innodb_stress failures.)

Reruns of aborted platforms:
https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-pipeline/5357/
https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-pipeline/5356/

kamil-holubicki commented on build-ps/debian/percona-xtradb-cluster-server-5.7.preinst:
```diff
@@ -1,7 +1,7 @@
 
 get_pcount () {
-	PSCOUNT=$(ps -ef | grep "/usr/sbin/mysqld" | wc -l)
+	PSCOUNT=$(ps -C mysqld --no-headers | wc -l)
```
I'm not sure if we should touch build-ps content. As per the process described in Conflu, let's leave this part for packaging team.

kamil-holubicki reviewed (commented):
As per commit message:
```
* Bumps the XB_REQUIRED_VERSION.
* Fixes trivial conflicts in build-ps/<PLATFORM>/control files.
```
If there are any conflicts, they should be resolved in the next step. 1st step should be merge with conflicts.
But, this does not apply to build-ps, as it should be kept as-is (pxc version)

XB_REQUIRED_VERSION should be bumped as conflict resolution commit (if there is no conflict, it can be done directly in PS merge commit)

In other words, I understand that it should be 1 commit where:
1. build-ps conflicts are ignored as we use PXC version
2. XB_REQUIRED_VERSION bumped, as there are no other conflicts

VarunNagaraju committed:
Implemented PXC-4132 (Merge PS-5.7.41, Part 2 no conflicts)

https://jira.percona.com/browse/PXC-4132

Merge tag 'Percona-Server-5.7.41-44' into PXC-4132-5.7.41-merge-part2

* Bumped the XB_REQUIRED_VERSION.
* Changes in build-ps/* files are ignored.
* Unnecessary build scripts remain deleted.

VarunNagaraju commented:
ASAN: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-pipeline/5374/

venkatesh-prasad-v reviewed (approved):
Please merge part-1 PR before merging this PR.
